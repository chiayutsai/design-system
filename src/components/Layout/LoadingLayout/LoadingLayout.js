import useStyles from 'isomorphic-style-loader/useStyles'
import Preloader from 'components/Preloader'
import styles from './LoadingLayout.scss'

const LoadingLayout = () => {
  useStyles(styles)

  return (
    <div className={styles.root}>
      <Preloader />
    </div>
  )
}

export default LoadingLayout
