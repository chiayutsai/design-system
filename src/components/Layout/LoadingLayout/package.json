{
  "name": "LoadingLayout",
  "version": "0.0.0",
  "private": true,
  "main": "./LoadingLayout.js"
}
