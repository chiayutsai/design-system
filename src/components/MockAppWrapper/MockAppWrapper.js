import history from 'history'
import PropTypes from 'prop-types'
import App from 'components/App'
import configureStore from 'store/configureStore'
import { injectReducer } from 'store/makeRootReducer'

const initialState = {}

export const store = configureStore(initialState, { history })

const MockAppWrapper = ({ children }) => (
  <App
    context={{
      pathname: '',
      store,
      injectReducer,
    }}
    insertCss={() => {}}>
    {children}
  </App>
)

MockAppWrapper.propTypes = {
  children: PropTypes.node.isRequired,
}

export default MockAppWrapper
