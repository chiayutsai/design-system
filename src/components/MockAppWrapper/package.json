{
  "name": "MockAppWrapper",
  "version": "0.0.0",
  "private": true,
  "main": "./MockAppWrapper.js"
}
