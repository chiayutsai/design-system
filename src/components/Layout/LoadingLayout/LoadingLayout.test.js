import renderer from 'react-test-renderer'
import MockAppWrapper from 'components/MockAppWrapper'
import LoadingLayout from './LoadingLayout'

describe('LoadingLayout', () => {
  test('render correctly', () => {
    const wrapper = renderer
      .create(
        <MockAppWrapper>
          <LoadingLayout />
        </MockAppWrapper>,
      )
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
