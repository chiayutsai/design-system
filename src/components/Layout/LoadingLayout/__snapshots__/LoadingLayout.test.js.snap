// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingLayout render correctly 1`] = `
<div
  className="root"
>
  <div
    className="preloader"
  >
    <div
      className="preloader_spiner"
    >
      <div
        className="preloader_circles"
      >
        <div
          className="preloader_circle-1"
        />
        <div
          className="preloader_circle-2"
        />
      </div>
    </div>
  </div>
</div>
`;
